dotnet/src/Agents/Abstractions/ChatHistoryChannel.cs

```diff
@@ -25,7 +25,7 @@ public class ChatHistoryChannel : AgentChannel
             throw new KernelException($"Invalid channel binding for agent: {agent.Id} ({agent.GetType().FullName})");
         }
 
-        await foreach (var message in historyHandler.InvokeAsync(this._history, cancellationToken).ConfigureAwait(false))
+        await foreach (var message in historyHandler.InvokeAsync(this._history, this.Logger, cancellationToken).ConfigureAwait(false))
```

Comment:
Can the `historyHandler` contain a log in itself?

Reply:
Good question.  `IChatHistoryHandler` is a adapter interface for the `Agent` concept and translates immediately into:

```c#
public class ChatHistoryAgent : KernelAgent, IChatHistoryHandler { ... }
```

Ultimately, the direction i've heading in this initial change is to _not_ assign logger to an agent.  This will be a good discussion point for tomorrow.

Reply:
I made that the agent (the implementation of `IHistoryHandler`) contains a logger so we don't need to pass in a logger in a method.